### PR TITLE
Bug 1868378 - Indicate the installation method when installing an add-on.

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1868378

### CODE REVIEW

File: android-components/components/browser/engine-gecko/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineTest.kt [hunk 1]
```diff
@@ -918,7 +919,7 @@ class GeckoEngineTest {
         var onErrorCalled = false
         val result = GeckoResult<GeckoWebExtension>()
 
-        whenever(extensionController.install(any())).thenReturn(result)
+        whenever(extensionController.install(any(), any())).thenReturn(result)
```
Comment (willdurand) [inline android-components/components/browser/engine-gecko/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineTest.kt:922]: we probably want to capture a `null` somewhere in this file (in an existing test) for when the install method is not specified.

File: android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/webextension/InstallationMethod.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.concept.engine.webextension
+
+/**
+ *  The method used to install a [WebExtension].
+ */
+enum class InstallationMethod {
+    /**
+     * Indicates the [WebExtension] was installed using from the UI add-ons manager.
```
Comment (willdurand) [inline android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/webextension/InstallationMethod.kt:12]:
```suggestion
     * Indicates the [WebExtension] was installed from the add-ons manager.
```

File: android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/webextension/WebExtensionRuntime.kt [hunk 3]
```diff
@@ -43,13 +43,15 @@ interface WebExtensionRuntime {
      * @param url the url pointing to an XPI file. An error is thrown when a resource URL is passed.
      * @param onSuccess (optional) callback invoked if the extension was installed successfully,
      * providing access to the [WebExtension] object for bi-directional messaging.
+     * @param installationMethod (optional) the method used to install a [WebExtension],
```
Comment (willdurand) [inline android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/webextension/WebExtensionRuntime.kt:46]:
```suggestion
     * @param installationMethod (optional) the method used to install a [WebExtension].
```

File: android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/AddonManager.kt [hunk 4]
```diff
@@ -125,18 +126,21 @@ class AddonManager(
      *  within the APK file (e.g. resource://android/assets/extensions/my_web_ext) or to a
      *  local (e.g. resource://android/assets/extensions/my_web_ext.xpi) or remote
      *  (e.g. https://addons.mozilla.org/firefox/downloads/file/123/some_web_ext.xpi) XPI file.
+     * @param installationMethod (optional) the method used to install a [Addon],
```
Comment (willdurand) [inline android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/AddonManager.kt:129]:
```suggestion
     * @param installationMethod (optional) the method used to install the [Addon].
```

File: android-components/components/feature/addons/src/test/java/mozilla/components/feature/addons/ui/AddonFilePickerTest.kt [hunk 5]
```diff
@@ -83,6 +83,6 @@ class AddonFilePickerTest {
 
         filePicker.handleUriSelected(uri)
 
-        verify(addonManager).installAddon(any<String>(), any(), any())
+        verify(addonManager).installAddon(any<String>(), any(), any(), any())
```
Comment (willdurand) [inline android-components/components/feature/addons/src/test/java/mozilla/components/feature/addons/ui/AddonFilePickerTest.kt:86]: Can we verify the install method here?

File: android-components/components/feature/addons/src/test/java/AddonManagerTest.kt [hunk 6]
```diff
@@ -574,6 +577,7 @@ class AddonManagerTest {
         )
 
         verify(engine).installWebExtension(
+            any(),
             any(),
```
Comment (willdurand) [inline android-components/components/feature/addons/src/test/java/AddonManagerTest.kt:581]: maybe we could `eq()` the install method too?

**Review by willdurand — APPROVED** [review]
r+wc, thanks

**Comment by Amejia481** [comment]
Thanks for the review, I update the patch with your comments

**Pull request merged**